lib/jellyfish/component.ex

```diff
@@ -44,14 +48,20 @@ defmodule Jellyfish.Component do
   @spec new(component(), map()) :: {:ok, t()} | {:error, term()}
   def new(type, options) do
     with {:ok, endpoint} <- type.config(options) do
+      metadata = get_metadata(type)
```

Comment:
Perhaps we could move the logic of getting metadata to the modules in question?
```suggestion
      metadata = type.metadata()
```